### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by develar** [develar/electron-complete-builder#3]

**Comment by ArekSredzki** [comment]
Definitely deb but rpm would be nice too.
I could really use this today, is it something you have experience with or shall I take a stab at it?
Thanks :)

**Comment by develar** [comment]
I don't know any issues in current Linux support (because electron-packager is used under the hood), only packaging in a distributable format is missed.

> I could really use this today

Or do you ask when `deb`-packaging will be supported?

**Comment by ArekSredzki** [comment]
Sorry for the ambiguity, I was referring to deb packaging.

**Comment by develar** [comment]
You can expect it in two days. My prototype works, but not yet production ready.

**Comment by develar** [comment]
@ArekSredzki Please try 0.5.2 Application installed to /opt/appName. Your feedback is important since my app doesn't has yet linux version and I cannot polish all aspects of deb package.

**Comment by ArekSredzki** [comment]
@devlar Thank you, I'll test this out within the hour and let you know.

**Comment by ArekSredzki** [comment]
I'm unable to open the i386 package, but here's my feedback for x64.

Ubuntu's software center indicates the following as qualities that make the package one of "bad quality":
 Lintian check results
 maintainer-address-missing XXXXXXX
 wrong-file-owner-uid-or-gid opt/ 1000/1000 <- repeated for every file in the package

**Comment by ArekSredzki** [comment]
That said, the package will install if the user ignores the warning, and can be opened from the /opt/appName directory like you mentioned. It would be nice if the installer linked the application so that users could search for it as an application.

**Comment by develar** [comment]
>   It would be nice if the installer linked the application so that users could search for it as an application.

Fixed in 0.5.4 (ubuntu mate: Applications -> Other -> YourName).

> Lintian

I have installed Lintian to check. I don't see this error in the new version, but see another warning. Could you please check 0.5.4 — is warning still appears?

**Comment by develar** [comment]
Forgot to mention: please do

`brew install ruby
gem install fpm`

on OS X to build. See https://github.com/jordansissel/fpm#system-packages

**Comment by ArekSredzki** [comment]
@develar Sorry for the slow follow up, I missed your message.

I've identified the issue as the following:
Lintian expects the maintainer address to contain an email address.
As such, if one enters an the package author as "Name Here <myemail@address.com>", lintian's requirements will be satisfied.
However, if you do this, the windows builds will fail as the field cannot contain the '@' character.

I tried the alternate npm form of entering the author as an object, such as:
```json
{
  "name": "Name Here",
  "email": "myemail@address.com"
}
```

But evidently you have not added support for this.

**Comment by ArekSredzki** [comment]
Also, currently the packaged linux app will only show the default os application icon, even though renderer windows define their own icons and work with when run with electron-prebuilt.
This likely has to do with fpm.

**Comment by develar** [comment]
@ArekSredzki As a quick workaround, specify `maintainer` explicitly: see https://github.com/develar/electron-complete-builder#distributable-format-configuration and [example](https://github.com/develar/electron-complete-builder/blob/master/test/fixtures/test-app-one/package.json#L17)

> will only show the default os application icon

I will fix it soon (see https://github.com/loopline-systems/electron-builder/issues/136). As a quick workaround, specify `desktopTemplate` (in the same way as `maintainer`) — it is a path to custom template.

**Comment by ArekSredzki** [comment]
@develar Thank you for the prompt responses!

Excited by the fast development of this project, and that it'll be merged in with electron-builder. :)

**Comment by develar** [comment]
@ArekSredzki It is already merged, I am preparing docs. Icons is fixed —version 0.6.3 published.

See updated README: "Linux icon set will be generated automatically on the fly from the OS X icns file."

and "Install required system packages:" (on Mac you need to install `brew install libicns`, on Linux 'sudo apt-get install icnsutils').

"Maintainer" issue will be fixed soon (in any case for now you can use suggested workaround).

**Comment by ArekSredzki** [comment]
@develar Much appreciated!

**Comment by develar** [comment]
Parsing author field was fixed, use 0.7.0.
